subprojects/gradle/kotlin-dsl-support/src/main/kotlin/com/avito/kotlin/dsl/TaskExtensions.kt

```diff
@@ -18,6 +18,11 @@ inline fun <reified T : Task> Task.dependencyOn(anotherTaskProvider: TaskProvide
     dependsOn(anotherTaskProvider)
 }
 
+inline fun <reified T : Task> Task.dependencyOn(anotherTask: T, configuration: (T) -> Unit) {
```

Comment:
Could we get a TaskProvider rather than Task?

Reply:
Why do I need TaskProvider if I already have a Task at hand?

Reply:
Only If we would use `named` (